File: leetcode2/2medium/노형민/Q2105.cs
public class Solution
{
    public int MinimumRefill(int[] plants, int capacityA, int capacityB)
    {
        int ap = 0;
        int bp = plants.Length - 1;

        int aw = capacityA;
        int bw = capacityB;

        int refills = 0;

        while (true)
        {
            if (ap > bp)
            {
                break;
            }
            if (ap == bp)
            {
                if (aw > bw && aw < plants[ap])
                {
                    refills++;
                }
                else if (bw > aw && bw < plants[bp])
                {
                    refills++;
                }
                else if (aw == bw && bw < plants[bp])
                {
                    refills++;
                }

                break;
            }

            if (aw < plants[ap])
            {
                refills++;
                aw = capacityA;
            }

            aw -= plants[ap];
            ap++;

            if (bw < plants[bp])
            {
                refills++;
                bw = capacityB;
            }

            bw -= plants[bp];
            bp--;
        }

        return refills;
    }
}

File: leetcode2/2medium/최원준/Q2105.java
package Leetcode.최원준;

/*
1. 아이디어 :


2. 시간복잡도 :
O( )

3. 자료구조/알고리즘 :

 */

public class Q2105 {
    class Solution {
        public int minimumRefill(int[] plants, int capacityA, int capacityB) {
            int n = plants.length;
            int A = capacityA, B = capacityB;
            int ans = 0;
            for (int i=0; i<n/2; i++) {
                if (plants[i] > A) {
                    ans++;
                    A = capacityA;
                }
                A -= plants[i];

                if (plants[n-i-1] > B) {
                    ans++;
                    B = capacityB;
                }
                B -= plants[n-i-1];
            }

            if (n%2 == 1) {
                int mid = n/2;
                if (A>=B) {
                    if (plants[mid] > A) ans++;
                } else {
                    if (plants[mid] > B) ans++;
                }
            }
            return ans++;
        }
    }
}
